test/e2e/data/infrastructure-docker/cluster-template.yaml

```diff
@@ -122,4 +122,39 @@ spec:
   unhealthyConditions:
     - type: E2ENodeUnhealthy
       status: "True"
-      timeout: 30s
\ No newline at end of file
+      timeout: 30s
+---
+apiVersion: addons.cluster.x-k8s.io/v1alpha3
+kind: ClusterResourceSet
+metadata:
+  name:  "${CLUSTER_NAME}-crs-0"
+spec:
+  clusterSelector:
+    matchLabels:
+      label: e2e
+  resources:
+    - name: test-secret1
+      kind: Secret
+    - name: test-configmap1
+      kind: ConfigMap
+    - name: test-notexist
+      kind: Secret
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: test-secret1
+type: clusterResourceSet
+stringData:
+  addon.yaml: |-
```

Comment:
I'm wondering if we should make a more realistic test and use ClusterResource test to apply the CNI for all our test clusters.
This is al little bit more invasive and requires some communications with the providers using the test framework, but IMO it allows us to exercise a more realistic use case instead of a completely fake one

Reply:
Can we address this in a separate PR?

Reply:
Yes, let's do this in a separate PR. Please file a follow-up issue.

Reply:
Created a follow-up PR: https://github.com/kubernetes-sigs/cluster-api/issues/3292